app/Dockerfile

```
@@ -0,0 +1,47 @@
+# from https://bun.sh/guides/ecosystem/docker, with modifications
+# to run a hot-reloading development server
+
+# use the official Bun image
+# see all versions at https://hub.docker.com/r/oven/bun/tags
+FROM oven/bun:1 AS base
+WORKDIR /app
+
+
+# -----------------------------------------------------------
+# install dependencies for dev and prod into temp directories
+FROM base AS install
+
+COPY package.json bun.lockb /temp/dev/
+RUN cd /temp/dev/ && \
+    bun install --frozen-lockfile
+
+# install with --production (exclude devDependencies)
+COPY package.json bun.lockb /temp/prod/
+RUN cd /temp/prod && \
+    bun install --frozen-lockfile --production
+
+# -----------------------------------------------------------
+# copy node_modules from dev stage, copy entire app
+# source into the image
+FROM base AS dev
+COPY --from=install /temp/dev/node_modules node_modules
+COPY . .
+# run the app in hot-reloading development mode
+# (this requires a new script, container-dev, in package.json that sets up vite
+# to accept connections on any interface, e.g. from outside the container, and
+# to always run on port 5713)
+CMD [ "vite", "--host", "--port", "5713" ]
+
+
+# -----------------------------------------------------------
+# copy production dependencies and source code into final image
+FROM base AS release
+COPY --from=install /temp/prod/node_modules node_modules
+COPY --from=prerelease /app/main.tsx .
+COPY --from=prerelease /app/package.json .
```

```suggestion
COPY --from=prerelease /app .
```

I think these copies need to be updated? So that vite build has access to all the source files to build them?

---

Err wait, should probably be just `COPY . .`

---

Duh, you're right; I'll just switch it to `COPY . .` as you suggested.